packages/vats/test/vpurse.test.js

```diff
@@ -170,7 +171,7 @@ test('makeVirtualPurse', async t => {
 
   expected.pushAmount(fungible837);
   await E(vpurse)
-    .deposit(payment, fungible837)
+    .deposit(payment, M.and(fungible837))
```

Comment:
> How does this fix #9407? I see only a refactor creating `AmountPatternShape` as an alias for `M.pattern()`.
> 
> Please include a test showing the behavior that failed before this PR and passes after.

Changing this like to provide another pattern, that in this case happens to mean the same thing, causes this test to fail before this PR with

```
  Error {
    message: 'In "retain" method of (VirtualPurseKit utils): arg 1?: "[match:and]" - Must be a copyRecord to match a copyRecord pattern: {"brand":"[match:remotable]","value":"[match:or]"}',
  }

  › async Object.receive (packages/vats/src/virtual-purse.js:219:23)
```
which points to the line
```js
          const amt = await this.facets.utils.retain(payment, optAmountShape);
```
because prior to this PR the retain guard was declared
```js
    retain: M.callWhen(PaymentShape).optional(amountShape).returns(amountShape),
```
and the pattern is not itself an amountShape.

Done. PTAL.